src/components/CCardDialog.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <v-dialog
+    v-model="model"
+    :width="width"
+    :max-width="maxWidth"
+    :min-width="minWidth"
+    :persistent="persistent"
+    class="c-dialog ma-2"
+  >
+    <c-card :title="title" :subtitle="subtitle" :loading="loading">
+      <template #prepend-actions>
+        <slot name="prepend-actions"></slot>
+      </template>
+
+      <template #actions>
+        <slot name="actions"></slot>
+      </template>
+
+      <slot></slot>
+    </c-card>
+  </v-dialog>
+</template>
+
+<script setup lang="ts">
+import CCard from "./CCard.vue";
+
+/**
+ * The model, whether the dialog should be shown or not.
+ */
+const model = defineModel<boolean>({ required: true });
+
+/**
+ * This component is a wrapper for the vuetify dialog with a card
+ * following the material design 3 specifications.
+ *
+ * @see https://m3.material.io/components/dialogs/overview
+ */
+withDefaults(
+  defineProps<{
+    /**
+     * The title of the dialog.
+     */
+    title?: string;
+
+    /**
+     * The subtitle of the dialog.
+     */
+    subtitle?: string;
+
+    /**
+     * The width of the dialog
+     */
+    width?: string;
+
+    /**
+     * The max width of the dialog.
+     */
+    maxWidth?: string;
+
+    /**
+     * The min width of the dialog.
+     */
+    minWidth?: string;
+
+    /**
+     * Whether clicking outside the element or pressing esc key will close the dialog.
+     */
+    persistent?: boolean;
+
+    /**
+     * Whether the loading bar should be shown or not.
+     */
+    loading?: boolean;
+  }>(),
+  {
+    title: undefined,
+    subtitle: undefined,
+    width: "500px",
+    maxWidth: undefined,
+    minWidth: undefined,
+    persistent: false,
+  },
+);
+</script>
+
+<style lang="scss">
+.c-dialog {
+  .v-overlay__content {
+    margin: 0 !important;
+    max-width: 100% !important;
```

Comment:
Because vuetify added a fixed margin (and idk why, but also the max-width being 100% - this margin) and doesn't reduce this on mobile. I removed it here and set our own responsive margin.